openvdb/io/Compression.cc

```diff
@@ -82,7 +82,9 @@ zipToStream(std::ostream& os, const char* data, size_t numBytes)
         os.write(reinterpret_cast<char*>(zippedData.get()), outZippedBytes);
     } else {
         // Write the size of the uncompressed data.
-        Int64 negBytes = -numBytes;
+        // numBytes expected to be <= the max value + 1 of a signed int64
+        assert(numBytes <= size_t(std::numeric_limits<Int64>::max())+size_t(1));
```

Comment:
Can you elaborate on the "+1" part of this?

Reply:
Sure, essentially size_t is a uint64_t on a lot of platforms (that we're targeting) but we negate the value to represent uncompressed data. This won't work if the value being passed in is greater than the max value of  **signed** 64 bit int (+1 as the lowest negative value will be one more than this) as it *may* (read: probably will) wrap and become positive again. In reality I sincerely much doubt anyone is writing out more than `9223372036854775807` bytes per leaf... but regardless this is still a potential bug. We're quite blasé about castinh between signed/unsigned values of the same precision with no checks and I'd like to try and improve these cases, especially as Windows complains at every single one

Reply:
Thanks. I suggest we just make this assert stricter than necessary here for greater simplicity (`numBytes < numeric_limits<Int64>::max()` ). I doubt anyone will see false positives as you say.